File: php/src/ApiInternal/AutoconfigSubscription.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\ApiInternal;

use Svix\Exception\ApiException;
use Svix\Models\AutoConfigSubscriptionOut;
use Svix\Request\SvixHttpClient;

class AutoconfigSubscription
{
    public AutoconfigSubscriptionDestination $destination;
    public AutoconfigSubscriptionEndpoint $endpoint;

    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
        $this->destination = new AutoconfigSubscriptionDestination($client);
        $this->endpoint = new AutoconfigSubscriptionEndpoint($client);
    }

    /**
     * Get an AutoConfig subscription, including the bound endpoint or destination if any.
     *
     * @throws ApiException
     */
    public function get(
        string $appId,
        string $autoconfigId,
    ): AutoConfigSubscriptionOut {
        $request = $this->client->newReq('GET', "/api/v1/app/{$appId}/autoconfig/{$autoconfigId}");
        $res = $this->client->send($request);

        return AutoConfigSubscriptionOut::fromJson($res);
    }
}

File: ruby/lib/svix/api_internal/autoconfig_subscription.rb
# frozen_string_literal: true
# This file is @generated

require "net/http"

module Svix
  class AutoconfigSubscription
    attr_accessor :destination
    attr_accessor :endpoint
    def initialize(client)
      @client = client
      @destination = AutoconfigSubscriptionDestination.new(client)
      @endpoint = AutoconfigSubscriptionEndpoint.new(client)
    end

    def get(app_id, autoconfig_id)
      res = @client.execute_request(
        "GET",
        "/api/v1/app/#{app_id}/autoconfig/#{autoconfig_id}"
      )
      AutoConfigSubscriptionOut.deserialize(res)
    end

  end
end
